lib/ex_aws/ecs.ex

```diff
@@ -0,0 +1,1091 @@
+  The `:maximum_percent` parameter represents an upper limit on the number of
+  running tasks during a deployment, which enables you to define the
+  deployment batch size. For example, if your service has a `:desired_count` of
+  four tasks, a `:maximum_percent` value of 200% starts four new tasks before
+  stopping the four older tasks (provided that the cluster resources required
+  to do this are available).
+
+  When `update_service/2` stops a task during a deployment, the equivalent of
+  `docker stop` is issued to the containers running in the task. This results
+  in a `SIGTERM` and a 30-second timeout, after which `SIGKILL` is sent and
+  the containers are forcibly stopped. If the container handles the `SIGTERM`
+  gracefully and exits within 30 seconds from receiving it, no `SIGKILL` is
+  sent.
+
+  When the service scheduler launches new tasks, it attempts to balance them
+  across the Availability Zones in your cluster with the following logic:
+
+  * Determine which of the container instances in your cluster can support your
+  service's task definition (for example, they have the required CPU, memory,
+  ports, and container instance attributes).
+  * By default, the service scheduler attempts to balance tasks across
+  Availability Zones in this manner (although you can choose a different placement strategy):
+    * Sort the valid container instances by the fewest number of running tasks for this service
+    in the same Availability Zone as the instance. For example, if zone A has one running service
+    task and zones B and C each have zero, valid container instances in either zone B or C are
+    considered optimal for placement.
+    * Place the new service task on a valid container instance in an optimal Availability Zone
+    (based on the previous steps), favoring container instances with the fewest
+    number of running tasks for this service.
+
+  When the service scheduler stops running tasks, it attempts to maintain balance across the
+  Availability Zones in your cluster using the following logic:
+
+  * Sort the container instances by the largest number of running tasks for this service in
+  the same Availability Zone as the instance. For example, if zone A has one running service
+  task and zones B and C each have two, container instances in either zone B or C are
+  considered optimal for termination.
+  * Stop the task on a container instance in an optimal Availability Zone
+  (based on the previous steps), favoring container instances with the largest
+  number of running tasks for this service.
+  """
+  @spec update_service(service :: binary, opts :: update_service_opts) :: ExAws.Operation.JSON.t
+  def update_service(service, opts \\ []) do
+    data = opts
+    |> normalize_opts
+    |> Map.merge(%{"service" => service})
+    request(:update_service, data)
+  end
+
+
+  ## Private parts
+
+  # Pascalize has room for improvements.. but it seems to work
+
+  defp pascalize_keys(opts) when is_map(opts) do
```

Comment:
I think we can simplify it significantly:
```
defp pascalize_keys(opts) do
  Map.new(opts, fn {k, v} -> {pascalize_key(k), v} end)
end

defp pascalize_key(key) do
  {first, rest} =
    key
    |> to_string
    |> Macro.camelize
    |> String.split_at(1)

  String.downcase(first) <> rest
end
```

I'd actually like to put this function in the `ExAws.Utils` module because I think it has applicability beyond this service as well.